packages/kbn-i18n/src/translation.ts

```diff
@@ -5,20 +5,39 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
+import { IntlShape, CustomFormats } from '@formatjs/intl';
 
-import { Formats } from './core/formats';
-
-export interface Translation {
+export interface TranslationInput {
   /**
    * Actual translated messages.
    */
-  messages: Record<string, string>;
+  messages: IntlShape['messages'];
   /**
    * Locale of the translated messages.
    */
-  locale?: string;
+  locale: IntlShape['locale'];
   /**
    * Set of options to the underlying formatter.
    */
-  formats?: Formats;
+  formats?: CustomFormats;
+}
+
+export interface Translation extends TranslationInput {
+  /**
+   * Actual translated messages.
+   */
+  messages: IntlShape['messages'];
+  /**
+   * Locale of the translated messages.
+   */
+  locale: IntlShape['locale'];
```

Comment:
Is it really necessary to redefine these two properties (`messages`, `locale`) if they are already defined in the parent interface?

Reply:
we do not expose the whole `IntlShape` for this interface, i guess we can use `Pick< IntlShape , 'locale' | 'messages'>` but I wanted to have them explicitly written here with the doc notes above them.
I can do otherwise if you feel strongly about it.

Reply:
Sorry I meant for the `extends TranslationInput` interface (above), which also defines them.

Reply:
Ah right. fixed